src/main/java/dk/kb/storage/util/DsStorageClient.java

```diff
@@ -33,16 +47,253 @@
+     * <p>
+     * Note: This method is not public as the returned {@code Iterator} is not closeable, which might lead to
+     * resource leaks in case of problems such as a remote caller disconnecting. This is to be handled outside
+     * of this context, i.e. in {@link #bytesToRecordStream}.
+     * @param byteStream JSON response with records from a ds-storage.
+     * @return an iterator of {@link DsRecordDto} de-serialized from the {@code byteStream}.
+     * @throws IOException if the {@code byteStream} could not be read.
+     */
+    static Iterator<DsRecordDto> bytesToRecordIterator(InputStream byteStream) throws IOException {
+        JsonFactory jFactory = new JsonFactory();
+        ObjectMapper mapper = new ObjectMapper();
+        jFactory.setCodec(mapper);
+        JsonParser jParser = jFactory.createParser(byteStream);
+
+        if (jParser.nextToken() != JsonToken.START_ARRAY) {
+            throw new IllegalStateException("Expected JSON START_ARRAY but got " + jParser.currentToken());
+        }
+
+        return new Iterator<>() {
+            private DsRecordDto nextRecord = null;
+            private boolean eolReached = false;
+
+            @Override
+            public boolean hasNext(){
+                ensureNext();
+                return nextRecord != null;
+            }
+
+            @Override
+            public DsRecordDto next() {
+                if (!hasNext()) {
+                    throw new IllegalStateException("next() called with hasNext() == false");
+                }
+                DsRecordDto record = nextRecord;
+                nextRecord = null;
+                return record;
+            }
+
+            /**
+             * Move to next JSON token. If it is an END_ARRAY, processing is stopped, else a DsRecordDto is read
+             */
+            private void ensureNext() {
+                if (nextRecord != null || eolReached) {
+                    return;
+                }
+                try {
+                    if (jParser.nextToken() == JsonToken.END_ARRAY) {
+                        eolReached = true;
+                        jParser.close();
+                        byteStream.close();
+                        return;
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException("Unable to read next JSON token", e);
+                }
+                try {
+                    nextRecord = jParser.readValueAs(DsRecordDto.class);
+                } catch (IOException e) {
+                    throw new RuntimeException("Unable to read DsRecordDto Object from JSON stream", e);
+                }
+            }
+        };
+    }
+
+    /**
+     * {@link DsRecordDto} specific stream that gives access to the highest modification time for any record
+     * that will be in the stream (not just up to the current point). The highest modification time will always
```

Comment:
In your parenthesis you are mentioning a current point. I guess it's the current point in stream? One could mistake it for current point in time as we are also talking about modification times